### ISSUE TITLE
Stop spamming my contacts without my consent!

### ISSUE THREAD

**Issue by silverbucket** [sourcerer-io/sourcerer-app#414]
It is a very unethical thing to do, and shockingly arrogant for an open source project.

**Comment by ligi** [comment]
Yea - same here - just got a mail from a follower complaining about this. Was not aware this happens when signing in - really not cool ;-(

**Comment by hsanjuan** [comment]
Same here, how can I delete all my data from your platform?

**Comment by pujiaxun** [comment]
really angry!

**Comment by XenGi** [comment]
The consent was given by a very tiny checkbox which is on by default. I just checked after getting the spam myself. Maybe make the checkbox more visible and default to disable it.

**Comment by asvrada** [comment]
Same here. Although this repo is well made, the spamming literally destroy everything else. I am not recommending this to my friend and I will no longer use it. 

STOP IT.

**Comment by silverbucket** [comment]
Yeah it's a shame that clearly so much good work went into this project and yet someone thought that spamming everyone in the users github list by default, before the user even sees the page, was a good idea. Totally killed this project for me, there's no world in which I would allow an app to email *everyone* that is following me on github, about anything, ever. Ever heard of a share button?

**Comment by Ruxton** [comment]
It shouldn't even be a checkbox option. Let alone an intentionally deceptive one. Very weak.

**Comment by aweimeow** [comment]
I even don't notice there is a checkbox for this action, after I authorized with GitHub account, it started to scan my profile, and send out invitation automatically.
I realized this app send invitation after heard from my friend, very uncomfortable with send mail to my contact. You SHOULD remove this function, it's useless and offend.

**Comment by foursixnine** [comment]
I just got also a message from a colleague getting spammed by this app... :/ Was quite nice... until this. If you want to invite people to use your app, please make sure that you ASK THE USER, before peforming this kind of actions...

**Comment by boredstiff** [comment]
If you all would like to report this like I am about to do, you can do so here: https://github.com/contact/report-abuse?report=sourcerer-io

**Comment by boredstiff** [comment]
It's also really scummy that in your README, you don't list what your real intentions are: 

# Why do you need GitHub permissions?

We use emails to identify commit authorship, read orgs access to get list of public repositories that you've contributed to. You also need to grant access to read this public information from an organization.

**Comment by osilla** [comment]
Hi everyone. We apologize for the confusion our UX is causing.  To clear up any issues, we present the choice to notify your followers on signup when you choose to analyze your public repositories (see attached).  Regarding GitHub permissions, we do not use the GitHub API to read email addresses. Our tool only looks at publicly available addresses from the public repositories a user submits and then notifies some of them should the user not optout.

We take your feedback seriously and are looking into ways to make this option more apparent.

<img width="976" alt="screen shot 2018-11-21 at 8 34 16 am" src="https://user-images.githubusercontent.com/20307788/48917092-dd691c80-ee39-11e8-9d3e-ac1a76094cee.png">

**Comment by foursixnine** [comment]
It took you 7 days to take this seriously?... That’s sad. Please, instead of making it par of the sign up workflow, make it an explicit user action. I normally don’t want to send invitations to people, for services that I use. Not even facebook.

**Comment by boredstiff** [comment]
I am not someone who connected an account, so why do I have to get spam email because someone else doesn't pay attention?

- It's apparently not clear enough for people that don't pay attention
- Why should a user whom I've never contributed toward have access to spam me?
- I, as someone who didn't (and won't) connect my account, did not give permission for you to have my email.
- Our tool only looks at publicly available addresses from the public repositories a user submits and then notifies some of them should the user not optout: I have _never_ interacted with the user who spammed me. So this can't be true. The closest degree that we may have interacted is that I may have interacted in a repository in which they also interacted via a third party's repository
- Your email has an "Unsubscribe" link at the bottom of it, so you're doing _something_ with it, you're storing it somewhere.

The bottom line is: don't spam people that aren't asking for it. It's extremely unethical and it turns people off toward your application. On the surface, it does look like a very neat app and one that I would have been interested in using for myself - I love looking at data like this. But because of this practice, I will instead warn people away from it at this point.

**Comment by osilla** [comment]
hi @foursixnine i'm so sorry for the delayed response.  It's really unacceptable of us.

**Comment by osilla** [comment]
hi @boredstiff thank you for this.  Please know that we are looking into this, both from the UX perspectives as well as the email associations our algos determine.

**Comment by silverbucket** [comment]
@osilla What do you mean "looking into this" ? Just remove it. Add a "share" button if people like it they can share. Don't email peoples contacts just to promote your app before the user even decides if they like it.

**Comment by osilla** [comment]
hi @silverbucket fair point.

**Comment by osilla** [comment]
hi @hsanjuan This thread got so long, I missed seeing your request.  Very sorry about that.  A user can delete all their data under Account in Settings.  

Here's the link to the page:
https://sourcerer.io/settings#account

**Comment by foursixnine** [comment]
@osilla Did you guys ever do anything to fix this?. Did you manage to implement @silverbucket's suggestion?. Or simply are just waiting it out?. Not cool.

**Comment by osilla** [comment]
hi @foursixnine - we have not been waiting it out. we have implemented a number of measures regarding user acquisition and more are in progress.

**Comment by XenGi** [comment]
Could we all calm down a bit? wtf...
They did something that the community doesn't like. They listened and acted. Now let them do there work and stop bashing. A more civilized tone would be very appreciated.
Some people in here act like children..

**Comment by sergey48k** [comment]
Closing this as not relevant any more. We no longer provide this feature.
